certbot-ci/certbot_integration_tests/utils/certbot_call.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+from __future__ import absolute_import
+from distutils.version import LooseVersion
+import subprocess
+import sys
+import os
+
+from certbot_integration_tests.utils import misc
+
+
+def certbot_test(certbot_args, directory_url, http_01_port, tls_alpn_01_port,
+                 config_dir, workspace, force_renew, capture_stdout):
+    new_environ = os.environ.copy()
+    new_environ['TMPDIR'] = workspace
+
+    additional_args = []
+    if misc.get_certbot_version() >= LooseVersion('0.30.0'):
+        additional_args.append('--no-random-sleep-on-renew')
+
+    if force_renew:
+        additional_args.append('--renew-by-default')
+
+    command = [
+        'certbot',
+        '--server', directory_url,
+        '--no-verify-ssl',
+        '--http-01-port', str(http_01_port),
+        '--https-port', str(tls_alpn_01_port),
+        '--manual-public-ip-logging-ok',
+        '--config-dir', config_dir,
+        '--work-dir', os.path.join(workspace, 'work'),
+        '--logs-dir', os.path.join(workspace, 'logs'),
+        '--non-interactive',
+        '--no-redirect',
+        '--agree-tos',
+        '--register-unsafely-without-email',
+        '--debug',
+        '-vv'
+    ]
+
+    command.extend(certbot_args)
+    command.extend(additional_args)
+
+    subprocess_call = subprocess.check_output if capture_stdout else subprocess.check_call
+
+    print('--> Invoke command:\n=====\n{0}\n====='.format(subprocess.list2cmdline(command)))
+    return subprocess_call(command, universal_newlines=True,
+                           cwd=workspace, env=new_environ)
+
+
+def main():
+    args = sys.argv[1:]
+
+    # Default config is pebble
+    directory_url = os.environ.get('SERVER', 'https://localhost:14000/dir')
+    http_01_port = int(os.environ.get('HTTP_01_PORT', '5002'))
+    tls_alpn_01_port = int(os.environ.get('TLS_ALPN_01_PORT', '5001'))
```

Comment:
nit: Can we reuse these URL and port values in a global so if they ever change, we're not making modifications to multiple places?

Reply:
I wanted to do that initially, but these variables are set only when the script is invoked from the shell, not during integration tests, and if they are global, they would be evaluated all the time. 

If your concern is about avoiding duplication, why not put this logic in a dedicated function with a meaningful name?

Reply:
This is a response to https://github.com/certbot/certbot/pull/7073/files#r289605994.

To clarify, what I wanted to add constants for were the default values like `https://localhost:14000/dir` which is used in multiple places. I don't have any complaints about the environment variable stuff here.

Reply:
Ok, you meant constants to share in all the project. Added a specific module for that, took the occasion also to put boulder directory urls and traefik api port in it.